config/v3_3_experimental/types/luks.go

```diff
@@ -50,8 +50,12 @@ func (l Luks) Validate(c path.ContextPath) (r report.Report) {
 	}
 
 	// fail if a key file is provided and is not valid
-	if err := validateURLNilOK(l.KeyFile.Source); err != nil {
-		r.AddOnError(c.Append("keys"), errors.ErrInvalidLuksKeyFile)
+	sources := l.KeyFile.GetSources()
+	for _, src := range sources {
```

Comment:
nit: let's drop the `sources` variable assignment and in-line it in the `if` as it isn't re-used.